ext/dom/xpath.c

```diff
@@ -446,6 +446,60 @@ PHP_METHOD(DOMXPath, registerPhpFunctionNS)
 	);
 }
 
+/* {{{ */
+PHP_METHOD(DOMXPath, quote) {
+	char *input;
+	size_t input_len;
+	smart_str output = {0};
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s", &input, &input_len) ==
+			FAILURE) {
+		RETURN_THROWS();
+	}
+	if (memchr(input, '\'', input_len) == NULL) {
+		smart_str_appendc(&output, '\'');
+		smart_str_appendl(&output, input, input_len);
+		smart_str_appendc(&output, '\'');
+	} else if (memchr(input, '"', input_len) == NULL) {
+		smart_str_appendc(&output, '"');
+		smart_str_appendl(&output, input, input_len);
+		smart_str_appendc(&output, '"');
+	} else {
+		// need to use the concat() trick published by Robert Rossney at https://stackoverflow.com/a/1352556/1067003
+		smart_str_appends(&output, "concat(");
+		for (size_t i = 0; i < input_len;) {
+			uintptr_t bytesUntilSingleQuote =
```

Comment:
You don't have to cast to uintptr_t. You can keep using const char* and substract pointers from each other. When 

So in other words:
`ptr_a - ptr_b` is the same as `(uintptr_t)ptr_a - (uintptr_t)ptr_b`.

Reply:
Also, I would prefer snake_case as that's usually the style in php internal code.

Reply:
re-check the code now